libs/ddhub-client-gateway-user-roles/src/lib/service/user-auth.service.ts

```diff
@@ -0,0 +1,70 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { SecretsEngineService } from '@dsb-client-gateway/dsb-client-gateway-secrets-engine';
+import { ConfigService } from '@nestjs/config';
+import {
+  AuthTokens,
+  UserRolesTokenService,
+  UserTokenData,
+} from './user-roles-token.service';
+import { UserRole } from '../const';
+
+@Injectable()
+export class UserAuthService {
+  protected readonly logger = new Logger(UserAuthService.name);
+
+  constructor(
+    protected readonly secretsEngineService: SecretsEngineService,
+    protected readonly userRolesTokenService: UserRolesTokenService,
+    protected readonly configService: ConfigService
+  ) {}
+
+  public verifyToken(accessToken: string): UserTokenData {
+    return this.userRolesTokenService.verifyToken(accessToken);
+  }
+
+  public isAuthEnabled(): boolean {
+    return this.configService.get('USER_AUTH_ENABLED', false);
+  }
+
+  public refreshToken(refreshToken: string): AuthTokens {
+    const isAuthEnabled: boolean = this.configService.get<boolean>(
+      'USER_AUTH_ENABLED',
+      false
+    );
+
+    if (!isAuthEnabled) {
+      throw new Error('Auth not enabled');
+    }
+
+    return this.userRolesTokenService.refreshToken(refreshToken);
+  }
+
+  public async login(username: string, password: string): Promise<AuthTokens> {
+    const isAuthEnabled: boolean = this.configService.get<boolean>(
+      'USER_AUTH_ENABLED',
+      false
+    );
+
+    if (!isAuthEnabled) {
+      throw new Error('Auth not enabled');
+    }
+
+    const userPassword: string | null =
+      await this.secretsEngineService.getUserAuthDetails(username);
+
+    if (!userPassword) {
+      throw new Error('User does not exists or password is incorrect');
+    }
+
+    if (password === userPassword) {
+      const accountType =
+        username === UserRole.ADMIN ? UserRole.ADMIN : UserRole.MESSAGING;
```

Comment:
username is any username set by the client gw admin and must not equate to the user role.

Reply:
will do